src/components/article/CreateArticle.jsx

```diff
@@ -41,26 +34,24 @@ class CreateArticle extends React.Component {
     const articleslug = `/articles/${slug}`;
 
     if (typeof message === 'string') {
-      setTimeout(() => {
-        history.push(articleslug);
-      }, 1000);
+      history.push(articleslug);
     }
-  }
+  };
 
   render() {
     const { message } = this.state;
+    const { isLoggedIn } = this.props;
+
+    if (isLoggedIn === null) return null;
+    if (!isLoggedIn) return <Redirect to="/login" />;
```

Comment:
@codeshiffu, You would want to wrap Redirect inside of `BrowserRouter`

Reply:
@omokaro-faith thanks for taking the time to review. Can you please state the reason why that's necessary?

Here's [an example](https://reacttraining.com/react-router/web/example/auth-workflow) of using `<Redirect />` from the official `react-router` site